third_party/move/move-compiler-v2/src/pipeline/abort_analysis.rs

```diff
@@ -0,0 +1,176 @@
+//! Computes if a given code offset
+//! - may lead to an abort, or
+//! - leads to an abort, or
+//! - doesn't lead to an abort
+
+use abstract_domain_derive::AbstractDomain;
+use move_binary_format::file_format::CodeOffset;
+use move_model::model::FunctionEnv;
+use move_stackless_bytecode::{
+    dataflow_analysis::{DataflowAnalysis, TransferFunctions},
+    dataflow_domains::{AbstractDomain, JoinResult, Plus2},
+    function_target::{FunctionData, FunctionTarget},
+    function_target_pipeline::{FunctionTargetProcessor, FunctionTargetsHolder},
+    stackless_bytecode::Bytecode,
+    stackless_control_flow_graph::StacklessControlFlowGraph,
+};
+use std::{collections::BTreeMap, fmt::Display};
+
+/// top: maybe abort later or not
+/// true: definitely aborting later
+/// false: definitely not aborting later
+/// bot: neither aborting nor returning later
```

Comment:
`bot` corresponds to leading to an infinite loop (nontermination).  Let's think carefully about what we want that to mean.  In practice, it will lead to an abort due to running out of gas.

Reply:
deprecated